src/shim/preload_libraries.c

```diff
@@ -477,3 +480,70 @@ FILE *tmpfile(void) {
         free(name);
     return rv;
 }
+
+static void _convert_stat_to_stat64(struct stat* s, struct stat64* s64) {
+    memset(s64, 0, sizeof(*s64));
+
```

Comment:
Optional, *might* be worth using a macro to avoid possibility of copying the wrong field. e.g.

```
#define COPY_X(x) s64->##x = s->##x
COPY_X(st_dev);
COPY_X(st_ino);
...
#undef COPY_X
```

Reply:
Good idea, thanks!